core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -42,7 +42,19 @@ public void serialize(LongDecimalWithOverflowAndLongState state, BlockBuilder ou
             long overflow = state.getOverflow();
             long[] decimal = state.getDecimalArray();
             int offset = state.getDecimalArrayOffset();
-            VARBINARY.writeSlice(out, Slices.wrappedLongArray(count, overflow, decimal[offset], decimal[offset + 1]));
+            long[] buffer = new long[4];
+            long decimalLowBytes = decimal[offset + 1];
+            long decimalHighBytes = decimal[offset];
+            buffer[0] = decimalLowBytes;
+            buffer[1] = decimalHighBytes;
+            buffer[2] = overflow;
+            buffer[3] = count;
+            // if decimalHighBytes == 0 and count == 1 and overflow == 0 we only write decimalLowBytes (bufferLength = 1)
```

Comment:
You miss one important case: ```3 longs -> decimal[offset] == 0```
see https://github.com/trinodb/trino/pull/13573#discussion_r942932796

Reply:
You can implement it by:
```
// append low
buffer[0] = decimalLowBytes;
offset = 1;
// append high
buffer[offset] = decimalHiBytes;
offset = offset + (decimalHiBytes == 0? 0 : 1);
// append overflow, count
buffer[offset] = buffer[offset] | overflow;
buffer[offset + 1] = count;
offset = offset + (overflow == 0 & count == 1) ? 0 : 2
```